bin/lint

```diff
@@ -96,79 +114,93 @@ end
 #
 # We also need this value for cppcheck to find some system headers again, on platforms like macOS,
 # where the system headers aren't found at /usr/include.
-set system_hdrs (clang -H -E ../etc/hdrs.c 2>&1 | head -1 | sed -e 's/^\. //' -e 's/\/[^/]*$//')
+readonly system_hdrs="$(clang -H -E ../etc/hdrs.c 2>&1 | head -1 | sed -e 's/^\. //' -e 's/\/[^/]*$//')"
 
 # On macOS the system headers used by `clang` may not be rooted at /usr/include.
-set lint_args -I. -I$system_hdrs
+lint_args=( -I. -I"${system_hdrs}" )
 
 # This is needed with clang on macOS. Without it `cppcheck` fails with
 # `#error Unsupported architecture` from `#include <sys/cdefs.h>`.
-if test "$machine_type" = "x86_64"
-    set lint_args $lint_args -D__x86_64__ -D__LP64__
-end
+if [[ "${machine_type}" == "x86_64" ]]
+then
+    lint_args+=(  -D__x86_64__ -D__LP64__ )
+fi
 
-if type -q include-what-you-use
+if command -v include-what-you-use > /dev/null
+then
     echo
     echo ========================================
     echo Running IWYU
     echo ========================================
-    for c_file in $c_files
-        set mapping_file
-        switch $os_name
-            case Darwin FreeBSD OpenBSD
-                set mapping_file ../etc/iwyu.bsd.map
-            case Linux CYGWIN\*
-                set mapping_file ../etc/iwyu.linux.map
-        end
-        if set -q mapping_file[1]
+    typeset mapping_file=""
+    case "${os_name}" in
+        Darwin | FreeBSD | OpenBSD )
+            mapping_file="../etc/iwyu.bsd.map"
+            ;;
+        Linux | CYGWIN* )
+            mapping_file="../etc/iwyu.linux.map"
+            ;;
+    esac
+    for c_file in "${c_files[@]}"
+    do
+        if [[ "${#c_file}" -ne 0 ]]
+        then
             include-what-you-use -Xiwyu --transitive_includes_only \
-                -Xiwyu --mapping_file=$mapping_file \
+                -Xiwyu --mapping_file="${mapping_file}" \
                 --std=c99 -Wno-expansion-to-defined -Wno-nullability-completeness \
-                $lint_args (../scripts/extract_flags $project_file $c_file) $c_file 2>&1
-        else  # hope for the best
+                "${lint_args[@]}" "$(../scripts/extract_flags "${project_file}" "${c_file}")" "${c_file}" 2>&1
```

Comment:
No quotes around the `../scripts/extract_flags` subcommand. That's because in this instance we need the values to be split into individual tokens. Otherwise IWYU gets a single arg as if you had typed `"-Idir1 -Idir2"` rather than two flags if the quotes weren't present. In other words, while it is often a good idea, or even required, to quote var expansion and subcommands sometimes you actually want the tokenizing after the expansion. I'll fix when merging.